### ISSUE TITLE
trim() for Change Email Address in UCP and ACP

### ISSUE THREAD

**Issue by AndreRl** [mybb/mybb#3484]
Change Email Address has no trim, if you enter email with space on the end, error appears:

> The email address you entered is invalid. Please enter a valid email address.

Solution: add trim() when saving new email

Original thread: [trim() for Change Email Address in UCP](https://community.mybb.com/thread-220432.html)

**Comment by dvz** [comment]
#3088

**Comment by labrocca** [comment]
Lots of admin input don't trim.  I've already suggested a full trim on $mybb->input by default. In what situation would you not trim whitespace?

**Comment by euantorano** [comment]
@labrocca see https://github.com/mybb/mybb/issues/3088#issuecomment-381153148 from @dvz who is against trimming all input. I personally think it best to trim all input and extend the [`MyBB::$clean_variables`](https://github.com/mybb/mybb/blob/feature/inc/class_core.php#L115) array if there are really some inputs that need to not be trimmed.

**Comment by labrocca** [comment]
Meh, I understand his concern but trim baby trim! Annoyed me for a decade in ACP when using Users search and I copy paste some blankspace.  Please trim and save me from this agony!

**Comment by euantorano** [comment]
Any other thoughts @mybb/developers @mybb/sqa? I doubt trimming input will cause any problems, and if it does we can exclude fields specifically. The benefits outweigh any potential risk IMO.

**Comment by dvz** [comment]
I suggest going the UI route instead (`type="email"` and others, including custom types validation) since UI is where these mistakes are introduced.

**Comment by Sama34** [comment]
I agree forcing trim is not the best route. Ideally a way to bypass such force without much hassle should be included at least for developers using the `get_input()` method.

I agree with @dvz regarging improving the UI experience as well.

**Comment by Sama34** [comment]
Ideally, all `<input>` should be updated for the new types (`email`, etc) and let the browser handle these.

On the back end, the code should probably use stock PHP verification functions.

It is possible this is the case already in 1.9, but I will open just so we can make sure it does.

Regards.
